libzeth/core/field_element_utils.hpp

```diff
@@ -52,14 +52,15 @@ std::string bigint_to_hex(const libff::bigint<FieldT::num_limbs> &limbs);
 /// and we have `res_bigint = [Limb0, Limb1, Limb2, Limb3]`,
 /// where every Limb{i} is of type `mp_limb_t`,
 template<typename FieldT>
-libff::bigint<FieldT::num_limbs> bigint_from_bytes(
-    const uint8_t bytes[(FieldT::num_bits + 8 - 1) / 8]);
-
-/// Convert an hexadecimal string to a field element
-template<typename FieldT> FieldT field_element_from_hex(std::string field_str);
+libff::bigint<FieldT::num_limbs> bigint_from_hex(const std::string &hex);
```

Comment:
I am not sure the comments above are still applicable to the changes you have made to the encoding functions. If I get it correctly, what you will have if you pass: `input_bigint = [Limb0, Limb1, Limb2, Limb3]` to your encoding function  `bigint_to_hex` and the result you should get is:
```
res = [x7, ..., x0, x15, ..., x8, x23, ..., x16, x31, ..., x24]
            Limb3       Limb2           Limb1         Limb0
```

i.e. the byte endianness is switched compared to what's written in the comment -> I preserved the byte endianness in the previous encoding by doing right shifts to select the bytes https://github.com/clearmatics/zeth/blob/develop/libzeth/libsnark_helpers/debug_helpers.tcc#L53-L54

Reply:
This should be fine because you used both "reverse" functions in your encoding "from" and "to" hex, so you have inverse functions. However, it is worth reflecting these changes to the comments otherwise this is confusing.

Reply:
I hadn't intended to change the encoding here.  I'll check this and fix it.

Reply:
(As discussed, I realised the new code was equivalent to the old code afterall.  Updated the comments a little bit.)

Reply:
Thanks